src/sentry/ratelimits/redis.py

```diff
@@ -25,22 +49,49 @@ def validate(self):
     def is_limited(self, key, limit, project=None, window=None):
         if window is None:
             window = self.window
-
-        key_hex = md5_text(key).hexdigest()
-        bucket = int(time() / window)
-
-        if project:
-            key = f"rl:{key_hex}:{project.id}:{bucket}"
-        else:
-            key = f"rl:{key_hex}:{bucket}"
+        redis_key = self._construct_redis_key(key, project=project, window=window)
 
         try:
             with self.cluster.map() as client:
-                result = client.incr(key)
-                client.expire(key, window)
+                result = client.incr(redis_key)
+                client.expire(redis_key, window)
```

Comment:
I think you'll have less redis usage if you do
```py
client.expire(redis_key, window - time() % window)
```
Otherwise you'll be extending the expire bucket's TTL into the next bucket's timeframe and it'd never be incremented.

Reply:
Does redis handle non-integer expiration times okay?

Reply:
`window - time() % window` will always be an integer.